### DISCUSSION TITLE
Whisper Api and Library

### DISCUSSION THREAD

**Discussion by ROHIT-ROSHAN** [openai/whisper#1427] [Show and tell]
How different is this from using api... and why is the api charged and not this?

**Comment by ryanheise** [comment]
Regarding the price of the API service, you don't have to pay for that if you wanted to just download the software and set it up yourself on your own machine or in the cloud, but then you would still either have to pay for your own hardware or rent that hardware in the cloud. So the question is, do you want to pay for your own hardware and electricity, and also spend the time to install the software and tune it to your application, or do you want to pay OpenAI for a service that is already deployed and ready to go? If you're developing an application that needs Whisper to perform transcription, some developers will go with OpenAI's ready-to-use API because it saves hassle and scales to support heavy loads, while other developers will prefer to spend a bit more of their own time setting up their own deployment and paying their own costs since they may be able to save money that way.

As for the differences between Whisper and the API, the main difference is that Whisper contains the latest features, including features contributed by members of the wider open source community. And since it's open source, if there's a feature you want or there's a bug you want fixed, it is possible for you to contribute to Whisper by adding that feature or fixing that bug yourself and contributing it back to the project. If you use the API, you don't get the latest features, and you can't add missing features yourself or fix bugs yourself.
